lib/location/cloud_service/cloud_service_here_rest.c

```diff
@@ -52,7 +52,7 @@ BUILD_ASSERT(sizeof(CONFIG_LOCATION_SERVICE_HERE_API_KEY) > 1,
 
 #define HTTP_REQUEST_BODY						\
 	"{"								\
-		"\"lte\":["						\
+		"\"lteCatM\":["						\
```

Comment:
I'm not sure we can do this. As you've said, it's not done according to specs.

> Better solution would be to format the payload based the current network mode, but lte_lc_cells_info -struct does not include that information.

You could add it there if you wanted to. LTE LC already has it internally in the `prev_lte_mode` variable.

We had thought of this earlier but skipped as the advantages hadn't been confirmed with the cloud services.

Reply:
Right, that should be a rather small change.
But I assume it still can't be merged as it's not following specs and the benefits are quite minimal?

Reply:
I'm not sure what you are referring to now with "it still can't be merged".
Adding network mode into `lte_lc_cells_info` structure is fine and its use in this file.

Reply:
Sending cells according to the current system in either `lteCatM` or `nbIot` object would be fine and according to the specs, right? However, it seems that some of the objects are named differently for LTE(-M) and NB-IoT, for example `rsrp` and `nrsrp`, so this needs to be taken into account in the implementation.

Reply:
Yes, that should be correct. Though I wonder when they changed the API, and what is the purpose of having the 'n' in `nrsrp`. On a side note, the example in the API documentation uses same cell data for LTE-M and NB-IoT, which I assume are not real measurements for NB-IoT.

Reply:
Yes, the examples seem to contain identical data, the object names are just different. I don't know why they have decided to use different naming, it seems unnecessary and only complicates the code.

Reply:
I noticed LWM2M object /4/0/2 specifies NRSRP for NB-IoT signal strength, and the referred 3GPP TS 36.133 uses NRSRP for NB-IoT, so it seems it's best to stick with that.